### PR TITLE
Adds XYAxis Label Rotation

### PR DESCRIPTION
**Pull request by mlzummo**
Fixes GH-597

Signed-off-by: Mathew Zummo <mzummo@gmail.com>

### CODE REVIEW

**Comment by mlzummo** [comment]
http://rawgit.com/mlzummo/Visualization/XYAXIS_LABEL_ROTATION_3_INVERSE/demos/dermatology.html?src/chart/Column.long-label

File: test/chartFactory.js [hunk 1]
```diff
@@ -16,6 +16,17 @@
                     );
                 });
             },
+            "long-label": function (callback) {
+                require(["test/DataFactory", "src/chart/Column"], function (DataFactory, Column) {
+                    callback(new Column()
+                        .columns(DataFactory.ND.subjects.columns)
+                        .data(DataFactory.ND.subjects["data-long-label"])
```
Comment (GordonSmith) [inline test/chartFactory.js]: FYI - You can inline the data here - you only want it in the factory file if its shared between two samples.

File: src/chart/XYAxis.css [hunk 2]
```diff
@@ -3,6 +3,10 @@
   font: 10px sans-serif;
 }
 
+.chart_XYAxis .tick > text {
+  font: 10px sans-serif;
+}
```
Comment (GordonSmith) [inline src/chart/XYAxis.css]: Is this needed?

File: src/chart/XYAxis.js [hunk 3]
```diff
@@ -359,19 +360,53 @@
...
                     .attr("visibility", function (d, i) { return i % margin.overlapModulus ? "hidden" : null; })
+                    .attr("transform", "rotate(0)")
+                ;
+                break;
+            case "rotate" :
+                var deg = -(this.xAxisLabelRotation()) || 0;
+                xAxis.selectAll(".tick > text")
+                    .each(function() {
+                        var elm = d3.select(this);
+                        if (deg !== 0 && margin.overlapModulus > 1) {
```
Comment (GordonSmith) [inline src/chart/XYAxis.js]: This test can be done outside the "each" which will then let you conditionally fall through to the default case when modulus === 0 or deg === 0.

**Comment by GordonSmith** [comment]
Test page seems to work well (there is an issue with the margin calculation, but its not really related to your change)

File: src/chart/XYAxis.css (hunk 2 above)
Comment (mlzummo) [inline src/chart/XYAxis.css]: I can remove it ... but i put it there to fix a bug i noticed i can't recall what the bug was 100% ... i believe  it was due to some css overwriting it cause the font size wasnt specific enough to the tick text

Comment (mlzummo) [inline src/chart/XYAxis.css]: it allows you to set the label font size as well ... so keep it?

**Comment by mlzummo** [comment]
@GordonSmith i noticed this yesterday with the margin:

https://github.com/hpcc-systems/Visualization/issues/1012

**Pull request merged**